### PR TITLE
#389 make buttons responsive

### LINKED ISSUE #389: Detail view bottom button text is too long
Having `Save & Confirm Allocation` etc on the button is very long when the screen size is small and the overflow doesn't look nice.

Think having the buttons shrink earlier will be good to solve this more easily

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #389 

- Made the buttons shrink slightly earlier
image
- Went off road and made the status crumb component just a single crumb when the screens too small:
image

which is trying to handle this situation

image

(And as I look at this, I notice we will need to also sort out the normal breadcrumbs for when we add the invoice number)

Not ideal, but more ideal than taking more vertical space by adding even more to the footer

### CODE REVIEW

File: packages/common/src/styles/theme.ts [hunk 1]
```diff
@@ -79,10 +79,10 @@ const themeOptions = {
...
     values: {
       xs: 0,
-      sm: 600,
-      md: 1024,
-      lg: 1440,
-      xl: 1536,
+      sm: 599,
+      md: 1025,
+      lg: 1441,
+      xl: 1537,
```
Comment (joshxg) [inline packages/common/src/styles/theme.ts:85]: when I use `.down` then the value is exclusive - so `down(1024)` is false and so I've been using `smallScreen` as `1024` and below which was using `down('md')`.. I can see how this  could be confusing so if you have any better ideas 🤷

File: packages/common/src/ui/components/domain/StatusCrumbs/StatusCrumbs.tsx [hunk 2]
```diff
@@ -42,6 +50,30 @@ export const StatusCrumbs = <StatusType extends string>(
...
+    Crumbs = statuses.map(status => {
+      const date = statusLog[status];
+
+      return (
+        <StyledText color={date ? 'secondary' : 'gray.main'} key={status}>
+          {t(statusFormatter(status))}
+        </StyledText>
+      );
+    });
+  }
```
Comment (joshxg) [inline packages/common/src/ui/components/domain/StatusCrumbs/StatusCrumbs.tsx:75]: can see that this looks a bit ugly. Though, I think it's OK as a solution 🤷

File: packages/common/src/hooks/useIsScreen/useIsExtraLargeScreen.ts [hunk 3]
```diff
@@ -1,7 +1,6 @@
-import { useMediaQuery, useTheme } from '@mui/material';
+import { useTheme, useMediaQuery } from '@mui/material';
 
 export const useIsExtraLargeScreen = (): boolean => {
   const theme = useTheme();
-
   return useMediaQuery(theme.breakpoints.up('xl'));
```
Comment (mark-prins) [inline packages/common/src/hooks/useIsScreen/useIsExtraLargeScreen.ts:5]: this one should use `useIsScreen` as well?

File: packages/common/src/styles/theme.ts (hunk 1 above)
Comment (mark-prins) [reply packages/common/src/styles/theme.ts:85]: I'm already confused! why is `sm` `599` when `md` is `1025`?
I think having `down('md')` is fine - it is to me anyway. I justified it with thinking that smaller than medium is small ( and xtra small ).. similarly medium is 1025 and up. All nicely hidden behind hooks anyway, so people won't even need to think about it!

File: packages/common/src/ui/components/domain/StatusCrumbs/StatusCrumbs.tsx (hunk 2 above)
Comment (mark-prins) [reply packages/common/src/ui/components/domain/StatusCrumbs/StatusCrumbs.tsx:75]: looks fine to me 🤷

**Review by mark-prins — APPROVED** [review]
👍  Good catch! looks way better on a small screen!!

File: packages/common/src/hooks/useIsScreen/useIsExtraLargeScreen.ts (hunk 3 above)
Comment (joshxg) [reply packages/common/src/hooks/useIsScreen/useIsExtraLargeScreen.ts:5]: the confusion is at a maximum!

`useIsScreen` is using `down`, so `useIsLargeScreen` is `down, xl` - where as extra large is `up` - anything above xl..

File: packages/common/src/styles/theme.ts (hunk 1 above)
Comment (joshxg) [reply packages/common/src/styles/theme.ts:85]: ah yes i've changed it to 601 ..

😕

**Pull request merged**